dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/DefaultSecurityService.java

```diff
@@ -90,9 +63,9 @@
 
     private static final String RECAPTCHA_VERIFY_URL = "https://www.google.com/recaptcha/api/siteverify";
 
-    private final LoadingCache<String, Integer> CACHE_USERNAME_FAILED_LOGIN_ATTEMPTS = Caffeine.newBuilder()
-        .expireAfterWrite( LOGIN_LOCKOUT_MINS, TimeUnit.MINUTES )
-        .build( ( u ) -> 0 );
+    private static final int SECONDS_PER_MIN = 60;
+
+    private Cache userFailedLoginAttemptCache;
```

Comment:
I suggest using generics for Cache, i.e. Cache<T>. We can use generics from Caffeine, and you can do the unchecked conversion behind the redis implementation (instead of doing it everywhere/on the outside).